src/conn/conn_layered.c

```diff
@@ -48,8 +48,9 @@ __layered_get_disagg_checkpoint(WT_SESSION_IMPL *session, const char **cfg,
     if (page_log->pl_get_complete_checkpoint_ext == NULL)
         WT_ERR(ENOTSUP);
 
-    WT_ERR(page_log->pl_get_complete_checkpoint_ext(page_log, &session->iface,
-      complete_checkpoint_lsn, NULL, complete_checkpoint_timestamp, complete_checkpoint_metadata));
+    ret = page_log->pl_get_complete_checkpoint_ext(page_log, &session->iface,
+      complete_checkpoint_lsn, NULL, complete_checkpoint_timestamp, complete_checkpoint_metadata);
+    WT_RET_NOTFOUND_OK(ret);
```

Comment:
Should the ret == 0 case also doing err handling like freeing the page_log_name?

Reply:
I think you want to only call terminate when there is an error that is not WT_NOTFOUND?